
syntax = "proto3";

package test;

// Interface exported by the server.
service Hello 
{
    // Ping the server 
    rpc Ping (PingRequest) returns (PingResponse) {}

    // Shutdown the server
    rpc Shutdown (ShutdownRequest) returns (ShutdownResponse) {}
}

message PingRequest
{
    // empty message
}

message PingResponse
{
    string msg = 1;
}

message ShutdownRequest
{
    string reason = 1;
}

message ShutdownResponse
{
    string msg = 1;
}
